rust_src/src/lisp.rs

```diff
@@ -130,13 +141,31 @@ pub enum LispType {
     Lisp_Float = 7,
 }
 
+impl LispObject {
+    #[allow(unused_unsafe)]
+    pub unsafe fn get_type_unchecked(self) -> LispType {
+        let res = (self.to_raw() & !VALMASK) as u8;
+        unsafe { mem::transmute(res) }
+    }
+
+    #[inline]
+    pub unsafe fn get_untaggedptr_unchecked(self) -> * mut libc::c_void {
```

Comment:
Likewise here: whilst dereferencing the pointer would be unsafe, I think is safe too, unless I'm missing something.